### PR TITLE
Added development process 

### LINKED ISSUE #154: Add issue and pr template
#### I am submitting a -
* [x]  **Feature Request**
* [] **Bug Report**

#### Describe your bug/feature -
Currently there is no template for issue and pull request and so there are problems in a well formatted issue which can be easy to understand. So there must be a template for both.

#### Are you working on this issue? (Yes/No)
Yes

### PR DESCRIPTION
**Pull request by Ankitanvesh**
I have added a brief description of the development process so that the future contributions can be well structured and organised. Fixes #154

### CODE REVIEW

**Comment by tachyons** [comment]
Disagree with making Readme too big so that it will be difficult to find the relavant information

**Comment by Ankitanvesh** [comment]
Shall I put this in another document and put the link to it in the Readme.

**Comment by Ankitanvesh** [comment]
@armudgal pls review it. Considering the above comments I would like to know whether I should put whole
of the Development process in a document and then add a link to it from the readme.

**Comment by fpintos** [comment]
It is common in open source project to have a CONTRIBUTING.md file, which describes the development process.

File: README.md [hunk 1]
```diff
@@ -79,6 +79,81 @@ production:
...
+    git merge upstream/master
+
+You should not have merge conflicts in step 3, unless you’ve made changes to your local *master* branch directly (which you should not do).
+
+## Step 3: Create the feature branch in your local and github repos
+Reason: All of your development should occur in feature branches - ideally, one branch per issue solved.  This keeps your local *master* branch clean (reflecting *upstream* master branch), allows you to abandon development approaches easily if required (e.g., simply delete the feature branch), and also allows you to maintain multiple work-in-process branches if you need to (e.g., work on a feature branch while also working on fixing a critical bug - each in its own branch).
+
+1. Create a branch whose name contains the github issue number. For example, if you are going to work on issue (which is, say, a new feature for ‘forgot password’ management):
+
+        git checkout -b forgot-password
```
Comment (fpintos) [inline README.md]: The example here is missing the issue number mentioned above.

Comment (Ankitanvesh) [reply README.md]: @fpintos pls review the changes. I have added a CONTRIBUTING.md file and added a link to it from the README.md.

**Comment by armudgal** [comment]
I agree with @tachyons here, we don't need to lengthen the well defined README.md, We can have a small CONTRIBUTING.md.

Also, I think it's unnecessary to mention the steps on how to clone, how to make a branch, e.t.c. Let's keep the contribution guidelines short and to the point so that the developers can start hacking on the project as soon as possible. 
We can maybe close this PR and work on a new one. Thoughts? 
Also please go through #146

**Comment by Ankitanvesh** [comment]
Sounds cool to me then I will close this one and try to shorten the guidelines.

**Pull request closed without merging**